flink-queryable-state/flink-queryable-state-runtime/src/test/java/org/apache/flink/queryablestate/itcases/HAQueryableStateFsBackendITCase.java

```diff
@@ -18,28 +18,99 @@
 
 package org.apache.flink.queryablestate.itcases;
 
+import org.apache.flink.configuration.ConfigConstants;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.configuration.HighAvailabilityOptions;
+import org.apache.flink.configuration.QueryableStateOptions;
+import org.apache.flink.configuration.TaskManagerOptions;
+import org.apache.flink.configuration.WebOptions;
+import org.apache.flink.queryablestate.client.QueryableStateClient;
 import org.apache.flink.runtime.state.AbstractStateBackend;
 import org.apache.flink.runtime.state.filesystem.FsStateBackend;
+import org.apache.flink.test.util.MiniClusterResource;
 
+import org.apache.curator.test.TestingServer;
+import org.junit.AfterClass;
 import org.junit.BeforeClass;
-import org.junit.Rule;
+import org.junit.ClassRule;
 import org.junit.rules.TemporaryFolder;
 
 /**
  * Several integration tests for queryable state using the {@link FsStateBackend}.
  */
-public class HAQueryableStateFsBackendITCase extends HAAbstractQueryableStateTestBase {
+public class HAQueryableStateFsBackendITCase extends AbstractQueryableStateTestBase {
```

Comment:
I tend to agree. I created an updated version of this here: #5595 It's still using inheritance but I changed it such that the base class now always calls `setDetached(true)`. I don't want to go into refactoring those tests now because there is already enough to do. Plus, I did remove one layer of inheritance. 😅